### PR TITLE
Fix _isDefTermSession not propagating upon pane split/close

### PR DESCRIPTION
**Pull request by PankajBhojwani**
## Summary of the Pull Request
In #13560 we added a member to `Pane` that lets it know if it was spawned as a default terminal session, but did not propagate that value when the pane gets split or when the pane closes. This commit fixes that. 

## Validation Steps Performed
A session spawned by a def term invocation remembers it even as it goes through splits

### CODE REVIEW

File: src/cascadia/TerminalApp/Pane.cpp [hunk 1]
```diff
@@ -2510,6 +2510,11 @@ std::pair<std::shared_ptr<Pane>, std::shared_ptr<Pane>> Pane::_Split(SplitDirect
     // Clear out our ID, only leaves should have IDs
     _id = {};
 
+    // Make sure to propagate our _isDefTermSession value
+    WalkTree([&](auto p) {
+        p->_isDefTermSession = _isDefTermSession;
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:2515]: Wait, I don't know if this is correct... we don't want *all* children of a DefTerm pane to be considered DefTerm, right? Otherwise random new splits will inherit unexpected behavior...

Comment (DHowett) [inline src/cascadia/TerminalApp/Pane.cpp:2515]: Also, don't we need to bring the value back when you close a split and the Parent becomes a Leaf again?

And like, we need to correctly handle the state when you open a DefTerm, split it, and close the DefTerm. The remaining leaf should NOT have this set to true ..

**Review by DHowett — CHANGES_REQUESTED** [review]

File: src/cascadia/TerminalApp/Pane.cpp (hunk 1 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalApp/Pane.cpp:2515]: I must be misunderstanding something - shouldn't any split made on a def term pane result in everything below it also being def term? I assumed there was no way for a def term session to end up having a pane that is not def term

Comment (PankajBhojwani) [reply src/cascadia/TerminalApp/Pane.cpp:2515]: > Also, don't we need to bring the value back when you close a split and the Parent becomes a Leaf again?

The parent's `_isDefTermSession` doesn't get overwritten when a split happens! So we don't need to bring the value back, it is already there

> And like, we need to correctly handle the state when you open a DefTerm, split it, and close the DefTerm. The remaining leaf should NOT have this set to true ..

Ah, this was my misunderstanding, I thought any splits made while in a def term session should also be def term, can fix that

Comment (PankajBhojwani) [reply src/cascadia/TerminalApp/Pane.cpp:2515]: (Resolving this so the convo can continue in the other one)

Comment (PankajBhojwani) [reply src/cascadia/TerminalApp/Pane.cpp:2515]: > Also, don't we need to bring the value back when you close a split and the Parent becomes a Leaf again?

Oh I think I understand what you mean now....

Pane A is def term, and `_Split` is called -
- Pane A becomes a parent, Pane B inherits the control/profile from Pane A, and Pane C is the new child
- Pane B should be considered a def term session, Pane C should not

So, when `Close` is called on Pane B, A inherits the control/profile from C and should also inherit C's `isDefTermSession`

In other words, the `isDefTermSession` value should be migrated back and forth between children and parents whenever the control/profile is migrated

Comment (DHowett) [reply src/cascadia/TerminalApp/Pane.cpp:2515]: So here's how I think about it. A def term session is *a session that originated outside Terminal.* That is the defining characteristic -- it isn't something that can be inherited by something that comes from inside Terminal, because that would violate that invariant.

Comment (PankajBhojwani) [reply src/cascadia/TerminalApp/Pane.cpp]: Fixed :)

File: src/cascadia/TerminalApp/Pane.h [hunk 2]
```diff
@@ -214,21 +214,24 @@ class Pane : public std::enable_shared_from_this<Pane>
...
+    // <--- SECTION FOR PROPERTIES THAT NEED TO BE TRANSFERRED BETWEEN CHILD/PARENT PANES UPON SPLITTING/CLOSING -->
     std::shared_ptr<Pane> _firstChild{ nullptr };
     std::shared_ptr<Pane> _secondChild{ nullptr };
     SplitState _splitState{ SplitState::None };
     float _desiredSplitPosition;
+    winrt::Microsoft::Terminal::Control::TermControl _control{ nullptr };
+    winrt::Microsoft::Terminal::TerminalConnection::ConnectionState _connectionState{ winrt::Microsoft::Terminal::TerminalConnection::ConnectionState::NotConnected };
+    winrt::Microsoft::Terminal::Settings::Model::Profile _profile{ nullptr };
+    bool _isDefTermSession{ false };
+    // <--- END SECTION -->
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/Pane.h:229]:
```suggestion
#pragma region Properties that need to be transferred between child/parent panes upon splitting/closing
    std::shared_ptr<Pane> _firstChild{ nullptr };
    std::shared_ptr<Pane> _secondChild{ nullptr };
    SplitState _splitState{ SplitState::None };
    float _desiredSplitPosition;
    winrt::Microsoft::Terminal::Control::TermControl _control{ nullptr };
    winrt::Microsoft::Terminal::TerminalConnection::ConnectionState _connectionState{ winrt::Microsoft::Terminal::TerminalConnection::ConnectionState::NotConnected };
    winrt::Microsoft::Terminal::Settings::Model::Profile _profile{ nullptr };
    bool _isDefTermSession{ false };
#pragma endregion
```
nit: if you use `pragma region`, VS and VSCode let you expand/collapse it

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalApp/Pane.h (hunk 2 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalApp/Pane.h:229]: thanks!

**Review by zadjii-msft — APPROVED** [review]

**Review by lhecker — APPROVED** [review]

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
Thanks for this!
